Penteract/Assets/Scripts/Barrel.cpp

```diff
@@ -90,12 +102,17 @@ void Barrel::Update() {
 		}
 		else {
 			destroy = false;
-			GameplaySystems::DestroyGameObject(barrel->GetParent());
+			if (particlesShadow) {
+				GameplaySystems::DestroyGameObject(barrel->GetParent()->GetParent());				
+			}
+			else {
+				GameplaySystems::DestroyGameObject(barrel->GetParent());
+			}
 		}
 	}
 
 	if (!onFloor) {
-		if (barrelMesh->GetComponent<ComponentTransform>()->GetGlobalPosition().y > 0.25f) {
+		if (parentTransform->GetGlobalPosition().y > 2.0f) {
 			float3 barrelPos = parentTransform->GetGlobalPosition();
 			barrelPos += float3(0, -forceOfFall, 0);
 			parentTransform->SetGlobalPosition(barrelPos);
```

Comment:
PLEASE, on the else enforce that the y component of the global position is EXACTLY the threshold (2.f). If not, in some computer the black mark won't be seen on the floor